packages/browser/src/login/oidc/ClientRegistrar.ts

```diff
@@ -44,67 +61,99 @@ export default class ClientRegistrar implements IClientRegistrar {
     issuerConfig: IIssuerConfig
   ): Promise<IClient> {
     // If client secret and/or client id are stored in storage, use those.
-    const [
-      storedClientId,
-      storedClientSecret,
-      // storedClientName,
-    ] = await Promise.all([
-      this.storageUtility.getForUser(options.sessionId, "clientId", {
-        secure: false,
-      }),
-      this.storageUtility.getForUser(options.sessionId, "clientSecret", {
-        secure: false,
-      }),
-      // this.storageUtility.getForUser(options.sessionId, "clientName", {
-      //   // FIXME: figure out how to persist secure storage at reload
-      //   secure: false,
-      // }),
-    ]);
-    if (storedClientId) {
+    const [storedClientId, storedClientSecret, storedClientExpiry] =
+      await Promise.all([
+        this.storageUtility.getForUser(options.sessionId, "clientId", {
+          secure: false,
+        }),
+        this.storageUtility.getForUser(options.sessionId, "clientSecret", {
+          secure: false,
+        }),
+        this.storageUtility.getForUser(options.sessionId, "clientExpiresAt", {
+          secure: false,
+        }),
+      ]);
+
+    // Handle public client identifiers:
+    if (typeof storedClientId === "string" && isValidUrl(storedClientId)) {
       return {
         clientId: storedClientId,
-        clientSecret: storedClientSecret,
-        clientType: "dynamic",
+        clientExpiresAt: 0,
+        clientType: "solid-oidc",
```

Comment:
No more converting a solid-oidc client to a dynamic client on page reload; now it always stays as a solid-oidc client.